mkt/developers/tasks.py

```diff
@@ -361,9 +359,8 @@ def save_promo_imgs(obj, img_content):
     dirname = obj.get_promo_img_dir()
     destination = os.path.join(dirname, '%s' % obj.pk)
     remove_promo_imgs(destination)
-    resize_promo_imgs(
-        tmp_dst, destination, mkt.PROMO_IMG_SIZES,
-        set_modified_on=[obj])
+    kw = resize_promo_imgs(tmp_dst, destination, mkt.PROMO_IMG_SIZES)
```

Comment:
Add a comment explaining that this only works because you are calling it directly instead of delaying it. Otherwise you'd have to chain a `update_from_data` call